prov/slurm/actions.go

```diff
@@ -44,7 +46,7 @@ type actionOperator struct {
 	outputs             []string
 }
 
-func (o actionOperator) ExecAction(ctx context.Context, cfg config.Configuration, taskID, deploymentID string, action *prov.Action) (bool, error) {
+func (o *actionOperator) ExecAction(ctx context.Context, cfg config.Configuration, taskID, deploymentID string, action *prov.Action) (bool, error) {
 	log.Debugf("Execute Action:%+v with taskID:%q, deploymentID:%q", action, taskID, deploymentID)
 	var err error
 	o.client, err = GetSSHClient(cfg)
```

Comment:
```suggestion
	if o.client == nil {
		o.client, err = GetSSHClient(cfg)
	}
```

Could be done only once otherwise we can have some issues

Reply:
Same should be for `consulClient`

Reply:
Yes, my bad...